wordpress-mobile/WordPress-Android
# Issue/14128 fetch related posts on reader post details screen from vm + usecase

Task #14128

This PR extracts only the "related posts fetch logic" from the reader post details fragment to useCase and viewModel making this part of code testable.

PS:
1. It currently targets `issue/14128-reader-related-posts` which will be merged to develop after implementing remaining sub-tasks from the linked task.
2. I'll try to refactor "related posts display related logic" in the next sub-task where I'll update the "related posts" design.

To test:

Test 1
- Open a wp.com Reader post.
- Notice that related posts are displayed as before at the bottom of the Reader post details screen.

Test 2
- Open a non wp.com Reader post.
- Notice that related posts are not displayed at the bottom of the Reader post details screen.

Test 3
- Open a wp.com Reader post with scrollable content.
- Enable Aeroplane mode.
- Scroll to bottom.
- Notice that related posts fetch request fails silently (matches behaviour prior to refactoring) and related posts are not shown.

Tests included in `ReaderPostDetailViewModelTest`, `ReaderFetchRelatedPostsUseCaseTest`.

PR submission checklist:
